excel.go

```diff
@@ -0,0 +1,217 @@
+package main
+
+import (
+	"github.com/xuri/excelize/v2"
+)
+
+const (
+	user = iota
+	portal
+	previous
+	timestamp
+	numCols        int    = 4
+	automatedSheet string = "PasswordManager"
+)
+
+func getMCFinUsers(f *excelize.File, config *Portal) (map[string]string, error) {
+	rows, err := f.GetRows(config.SheetName)
+	if err != nil {
+		config.errorLog.Printf("failed getting rows from %s in %s", config.SheetName, config.Filename)
+		return nil, err
+	}
+
+	users := make(map[string]string)
+
+	headerToXCoord := getHeaderToXCoord(rows[0])
+	usernameXCoord := headerToXCoord[config.UsernameHeader]
+	passwordXCoord := headerToXCoord[config.PasswordHeader]
+
+	for _, row := range rows[rowOffset:] {
+		// check if row is empty
+		if row == nil {
+			continue
+		}
+		users[row[usernameXCoord]] = row[passwordXCoord]
+	}
+
+	return users, nil
+}
+
+func getPasswordManagerUsers(f *excelize.File, config *Portal) (map[string]string, error) {
+	rows, err := f.GetRows(automatedSheet)
+	if err != nil {
+		config.errorLog.Printf("failed getting rows from %s in %s", automatedSheet, config.Filename)
```

Comment:
a broader comment about these error logs: is it necessary to do ones like these at all? why not just do
```
return fmt.Errorf("failed getting rows from %s in %s: %s", automatedSheet, config.Filename, err)
```
and let the caller decide what to do with it?

There are a bunch of error logs like this.

Reply:
Using that pattern: Let the caller decide what to do with it. I've changed all the errors so they are returned to the caller.